plugins/server_links/ui/ServerLinks.component.ts

```diff
@@ -1,87 +1,101 @@
+    componentIds,
+    Grid,
+    type GridCellFunction,
+    centeredText,
+    leftAlignedText,
+    Paginator,
+} from '../../ui/UI.js'
+import type {ServerInfo} from '../ServerLinks.js'
 
 export default class ServerLinks extends StaticComponent {
 
-  private readonly paginator: Paginator
-  private readonly header: StaticHeader
-  private readonly grid: Grid
-  private serverList: ServerInfo[] = []
-  private readonly gameModeMap: { [gameMode in tm.GameMode]: string } = {
-    TimeAttack: 'TA',
-    Rounds: 'RND',
-    Teams: 'TEAM',
-    Cup: 'CUP',
-    Laps: 'LAP',
-    Stunts: 'STNT'
-  }
-  private readonly gameModeIcons: { [gameMode in tm.GameMode]: string } = config.icons.gameMode
-  private currentDefaultPage = 0 // Gets incremented to 1 at first update
-  private pageSwitchTimeouts: string[] = []
-
-  constructor() {
-    super(componentIds.serverLinks)
-    this.header = new StaticHeader('race')
-    this.grid = new Grid(config.width,
-      config.height - this.header.options.height,
-      [1], new Array(config.entries).fill(1))
-    this.paginator = new Paginator(this.id, 0, 0, 4)
-    this.paginator.onPageChange = (login) => {
-      const obj = this.displayToPlayer(login)
-      if (obj !== undefined) {
-        tm.sendManialink(obj.xml, obj.login)
-      }
+    private readonly paginator: Paginator
+    private readonly header: StaticHeader
+    private readonly grid: Grid
+    private serverList: ServerInfo[] = []
+    private readonly gameModeMap: { [gameMode in tm.GameMode]: string } = {
+        TimeAttack: 'TA',
+        Rounds: 'RND',
+        Teams: 'TEAM',
+        Cup: 'CUP',
+        Laps: 'LAP',
+        Stunts: 'STNT'
     }
-    this.onPanelHide((player) => {
-      this.sendMultipleManialinks(this.displayToPlayer(player.login))
-    })
-  }
-
-  getHeight(): number {
-    return config.height
-  }
+    private readonly gameModeIcons: { [gameMode in tm.GameMode]: string } = config.icons.gameMode
+    private currentDefaultPage = 0 // Gets incremented to 1 at first update
+    private pageSwitchTimeouts: string[] = []
 
-  update(newList: ServerInfo[]) {
-    this.serverList = newList
-    this.paginator.setPageCount(Math.ceil(this.serverList.length / config.entries))
-    this.currentDefaultPage++
-    if (this.currentDefaultPage > this.paginator.pageCount) {
-      this.currentDefaultPage = 1
-    }
-    for (const e of tm.players.list) {
-      if (this.pageSwitchTimeouts.some(a => a === e.login)) {
-        this.pageSwitchTimeouts = this.pageSwitchTimeouts.filter(a => a !== e.login)
-      } else {
-        this.paginator.setPageForLogin(e.login, this.currentDefaultPage)
-      }
+    constructor() {
+        super(componentIds.serverLinks)
+        this.header = new StaticHeader('race')
+        this.grid = new Grid(config.width,
+            config.height - this.header.options.height,
+            [1], new Array(config.entries).fill(1))
+        this.paginator = new Paginator(this.id, 0, 0, 4)
```

Comment:
I see that the paginator switches automatically but we could also add navigation buttons in the header (like in bestCps component). Its kinda complicated code and not really needed so we can also fix that later